Replaced ValueError with ZeroDivisionError

Giscash0 opened the pull request:
Creating a ZeroDivisionError was required by the specifications to handle cases where the denominator is zero

Lawr715 reviewed (commented):
I am pretty sure we do not need a separate class for ZeroDivisionError since it is not stated in the specs that we need to create a custom exception. From the way I see it, as long as it was raised then we should be good. Also the self._numerator and self._denominator from the version I committed earlier is the one needed for its immutability that will be done later on.

Lawr715 reviewed (commented):
I am pretty sure we do not need a separate class for ZeroDivisionError since it is not stated in the specs that we need to create a custom exception. From the way I see it, as long as it was raised then we should be good. Also the self._numerator and self._denominator from the version I committed earlier is the one needed for its immutability that will be done later on.

Lawr715 reviewed (commented):
I am pretty sure we do not need a separate class for ZeroDivisionError since it is not stated in the specs that we need to create a custom exception. From the way I see it, as long as it was raised then we should be good. Also the self._numerator and self._denominator from the version I committed earlier is the one needed for its immutability that will be done later on.

Giscash0 commented:
I removed the self implemented ZeroDivisionError. Also implemented the gcd function. It uses the euclidean algorithm taught in Module 1 of Math 51.4. We need to cite that.

Lawr715 reviewed (commented):
I believe there should be a return statement on the else logic off def gcd for the Fraction.gcd(b, a % b). Other than that, it looks good to me.
